### PR TITLE
1.0.0

### PR DESCRIPTION
**Pull request by tolgacanunal**

### CODE REVIEW

File: hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+    private val retrofitExceptionHandler =
+        RetrofitErrorHandler(
+            Gson(),
+            defaultErrorMessage,
+            intArrayOf(402)
+        )
+
+    @Before
+    fun setup() {
```
Comment (mitsinsar) [inline hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt:41]: JUnit4 creates a new class instance for per method. So, setup method should be static. Kotlin doesn't have the static keyword. That's why in JUnit5, they added new annotation called "TestInstance(Lifecycle.PER_CLASS)". By using this annotation, you tell JUnit5 to create just one instance of test class for whole methods.

https://medium.com/@m.sinan.sari/junit5-kotlin-7f937a6aef35

File: hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,244 @@
+package com.hipo.hipoexceptionsandroid
+
+import com.google.gson.Gson
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Before
+import org.junit.Test
+import retrofit2.Call
+import java.io.*
```
Comment (mitsinsar) [inline hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt]: Should we change wildcard import?

**Review by mitsinsar — APPROVED** [review]

File: hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt (hunk 1 above)
Comment (tolgacanunal) [reply hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt:41]: I'll switch to Junit5 later then. Thanks for letting me know.

File: hipoexceptionsandroid/build.gradle [hunk 3]
```diff
@@ -7,7 +7,7 @@ android {
 
 
     defaultConfig {
-        minSdkVersion 21
+        minSdkVersion 19
```
Comment (yasin-ce) [inline hipoexceptionsandroid/build.gradle]: We can make it 21. We won't have any apps below 21.

File: hipoexceptionsandroid/build.gradle [hunk 4]
```diff
@@ -23,10 +23,29 @@ android {
...
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "com.google.code.gson:gson:2.8.5"
+
+    implementation 'com.google.firebase:firebase-core:17.2.0'
+    implementation 'com.crashlytics.sdk.android:crashlytics:2.10.1'
+    implementation "com.squareup.retrofit2:retrofit:2.6.2"
+
+    testImplementation 'com.squareup.retrofit2:converter-gson:2.5.0'
+    testImplementation("com.squareup.okhttp3:mockwebserver:4.2.1")
     testImplementation 'junit:junit:4.12'
```
Comment (yasin-ce) [inline hipoexceptionsandroid/build.gradle]: We can create a versions file to stick with our app structure.

File: hipoexceptionsandroid/src/main/java/com/hipo/hipoexceptionsandroid/BaseErrorModel.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.hipo.hipoexceptionsandroid
+
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import com.google.gson.annotations.SerializedName
+
+data class BaseErrorModel(
```
Comment (yasin-ce) [inline hipoexceptionsandroid/src/main/java/com/hipo/hipoexceptionsandroid/BaseErrorModel.kt]: I think "Model" word is unnecessary here.

File: hipoexceptionsandroid/src/main/java/com/hipo/hipoexceptionsandroid/RetrofitErrorHandlerExceptionUtils.kt [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+class FallbackMessageException(requestDetail: String): Exception(requestDetail)
+class NoFallbackMessageException(requestDetail: String): Exception(requestDetail)
+class UnexpectedResponseCodeException(requestDetail: String): Exception(requestDetail)
+
+fun sendExceptionLog(exception: Exception) {
+    try {
+        Crashlytics.logException(exception)
+    } catch (exception: Exception) {
+        Log.w("RetrofitErrorHandler", "Crashlytics has been not started yet.")
```
Comment (yasin-ce) [inline hipoexceptionsandroid/src/main/java/com/hipo/hipoexceptionsandroid/RetrofitErrorHandlerExceptionUtils.kt]: We might create a const TAG here.

**Review by yasin-ce — APPROVED** [review]

**Pull request merged**